### PR TITLE
#184581340  User can change password

### PR DESCRIPTION
**Pull request by bencyubahiro77**
[Delivers #184581340]

#### What does this PR do?
The PR provide ability to the user to update their password

#### Description of Task to be completed?
the user should be able to change his password if he/she wants too. he will provide the old password , then new password and the new password provided should be able to match the criteria of a valid password.

#### How should this be manually tested?
this can be tested using postman where you can
PATCH users/change-password
-In the body insert
{
    "email": " ",
    "oldPassword": " ",
    "newPassword": " "
}

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184581340

#### Screenshots (if appropriate)
Screenshot 2023-03-14 at 18 51 26

### CODE REVIEW

**Review by edwinnambaje — APPROVED** [review]

**Review by edwinnambaje — CHANGES_REQUESTED** [review]
i see the route of changing password is not protected so someone who havent logged in will change password?

**Comment by bencyubahiro77** [comment]
> i see the route of changing password is not protected so someone who havent logged in will change password?

can you help me with that i have been stack

File: src/routes/users.js [hunk 1]
```diff
@@ -12,10 +12,13 @@ const routes = express.Router();
 
 routes.post('/signup', signupVatidate, checkUserExist, Users.signup);
 routes.get('/verify-email/:token', Users.verifyEmail);
+routes.patch('/change-password', Users.changePassword);
```
Comment (blackd44) [inline src/routes/users.js]: protect this route
make it like 
`routes.patch('/change-password', isAuthenticated, Users.changePassword);`

File: src/controllers/users.controllers.js [hunk 2]
```diff
@@ -197,4 +197,39 @@ export default class Users {
       res.status(500).json({ 'Error:': error });
     }
   }
+  static async changePassword(req, res) {
+    try {
+      const { email, oldPassword, newPassword } = req.body;
```
Comment (blackd44) [inline src/controllers/users.controllers.js]: if protected, the email won't be necessary
it will be `req.user.email`

**Review by blackd44 — APPROVED** [review]
other are good except those.

File: src/routes/users.js (hunk 1 above)
Comment (bencyubahiro77) [reply src/routes/users.js]: yes i am doing it the issues i am having problem with mocha

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**